### PR TITLE
Aarch64 pagetable

### PR DESCRIPTION
**Pull request by Skallwar**
Add support for aarch64 pagetable

### CODE REVIEW

**Comment by Skallwar** [comment]
The initialization on of the PMM is considerably slower (to the point where riscv64 ci timeouts). I think we should merge this first and try to optimize it later on

File: drivers/src/qemuexit.rs [hunk 1]
```diff
@@ -6,18 +6,26 @@ use crate::Driver;
...
+        return Self {
             address: RISCV64_BASE_ADDRESS,
             inner: qemu_exit::RISCV64::new(RISCV64_BASE_ADDRESS as u64),
-        }
+        };
+
+        #[cfg(target_arch = "aarch64")]
+        return Self {
+            inner: qemu_exit::AArch64::new(),
+        };
```
Comment (CohenArthur) [inline drivers/src/qemuexit.rs:28]: You don't have to use `return`s for this
```suggestion
        #[cfg(target_arch = "aarch64")]
        Self {
            inner: qemu_exit::AArch64::new(),
        }
```
(and similarly for the `riscv64` version)

should suffice. You can simply keep both expression blocks hidden behind `#[cfg]` directives unless I'm missing something

File: drivers/src/qemuexit.rs [hunk 2]
```diff
@@ -36,7 +44,11 @@ impl Default for QemuExit {
...
 
 impl Driver for QemuExit {
-    fn get_address_range(&self) -> (usize, usize) {
-        (self.address, 1)
+    fn get_address_range(&self) -> Option<(usize, usize)> {
+        #[cfg(target_arch = "riscv64")]
+        return Some((self.address, 1));
+
+        #[cfg(target_arch = "aarch64")]
+        return None;
```
Comment (CohenArthur) [inline drivers/src/qemuexit.rs:52]: Same here
```suggestion
        #[cfg(target_arch = "riscv64")]
        { Some((self.address, 1)) }

        #[cfg(target_arch = "aarch64")]
        { None }
```

File: kernel/src/arch/aarch64/pgt48.rs [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+    fn disable(&mut self) {
+        // let satp = Satp::new()
+        //     .with_ppn(0)
+        //     .with_asid(0)
+        //     .with_mode(SatpMode::Bare as u8);
+
+        // unsafe {
+        //     asm!("csrw satp, {}", in(reg)u64::from(satp));
+        //     asm!("sfence.vma");
+        // }
```
Comment (CohenArthur) [inline kernel/src/arch/aarch64/pgt48.rs:352]: Dead code? Or is that the valid version but kept away for now behind the `unimplemented!()`?

**Review by CohenArthur — APPROVED** [review]
LGTM other than some very small nitpicks :)

**Pull request merged**